lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -6456,6 +6454,49 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
     return errorOccurred;
   }
 
+  /// If original result is non-varied, it will always have a zero derivative.
+  /// Skip full pullback generation and simply return zero for wrt parameters.
+  void handleNonVariedResult(SILValue origNonvariedResult) {
```

Comment:
“Handle” is not a good name for methods as it does not at all describe what it does other than what it might take as an argument.

Reply:
How about `runShortcutGenerationForNonVariedResult`?

Reply:
"generateShortcut" also is also very unclear. When naming a method, I'd suggest starting with _clear words_ that describe exactly what the method does, before introducing any new terms like "shortcut" or using any obscure concepts.
In this case, I would go with something like `emitZeroDerivativesForNonvariedResult`. Also note that we rarely use the word "generate" in the AD code path — it would be good to unify them to be "emit".

Reply:
That makes sense, thanks! I'll keep this principle in mind 🙂 
Renamed to `emitZeroDerivativesForNonvariedResult` in https://github.com/apple/swift/pull/27554/commits/a7ab4d27bf5e43f6fd6a0e42b4e253a6df4b3570.